### ISSUE TITLE
[Feature] Add a flag to disable internet access on worker nodes

### ISSUE THREAD

**Issue by denismakogon** [oracle-terraform-modules/terraform-oci-oke#331]
When creating a cluster there is no way to disable internet access from inside worker nodes. This means that by default without a proper network policy (via calico) every pod provisioned on a worker node has direct access to the internet.

So, I'd like to see a boolean flag that will prevent the provisioning of an additional egress rule: `Destination: 0.0.0.0/0 on all protocols`.

Something like:
```terraform
variable "allow_worker_internet_access" {
  description = "allow worker nodes to start egress connection to internet"
  default     = true
  type        = bool
}
```

**Comment by hyder** [comment]
Will address this in #308

**Comment by hyder** [comment]
What is the use case btw? if you disable internet access, then you won't be able to pull images from external repositories e.g. dockerhub etc. The assumption is that you'll only be using OCIR. Is this based on a customer request?

**Comment by denismakogon** [comment]
Indeed, most of images, even community ones are copied to our own OCIR registry.

**Comment by denismakogon** [comment]
To be short, the only services that worker nodes need to access belong to OCI, the rest of outgoing internet connections are forbidden for all components of the deployment.
